vignettes/GMAT-GMST.Rmd

```diff
@@ -0,0 +1,114 @@
+---
+title: "Example: Calculating GMST from GMAT in Hector"
+author: "Leeya Pressburger"
+date: "2021-08-18"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteEncoding{UTF-8}
+  %\VignetteIndexEntry{Example: Carbon tracking with Hector}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Introduction
+
+The global mean surface temperature (GMST) is the average of the air temperature over land and the temperature at the surface of the ocean. This is a very useful metric for observing global warming or global cooling. However, Hector does not produce GMST as an output at this time. Hector does measure the global mean air temperature (GMAT), which we can use to back-calculate GMST. 
+
+This vignette will demonstrate how to set up the calculation between GMAT and GMST then plot results to observe differences over time. 
+
+## Setup
+
+Let's load the packages we will need and run a scenario.
+
+*** will have to recode without dplyr/tidyr - might need help with this
```

Comment:
👍

Reply:
Where I'm getting stuck:

> hector_output %>% 
    filter(variable %in% c(LAND_AIR_TEMP(), OCEAN_SURFACE_TEMP())) %>% 
    spread(variable, value) %>%
    # Calculate the area weighted sum of the land & ocean temperature
    mutate(Tgav_land = Tgav_land * flnd,
           Tgav_ocean_ST = Tgav_ocean_ST * (1 - flnd),
           value = (Tgav_land + Tgav_ocean_ST)) %>%
    select(year, value) %>% 
    mutate(variable = "GMST") -> global_surface_temp

The first line (filter) can become:
> hector_output <- subset(hector_output, variable == c(LAND_AIR_TEMP(), OCEAN_SURFACE_TEMP()))

But I'm not sure how to recode spread or mutate.

Reply:
`x %>% mutate(y = 1)` is just `x$y <- 1` in base R.

Instead of `spread` you can use `reshape`.